fms_extras/utils/generation.py

```diff
@@ -0,0 +1,141 @@
+import functools
+import time
+from typing import Any, Callable, List, MutableMapping, Optional, Union
+
+import torch
+import torch.nn.functional as F
+
+from fms_extras.utils.cache.paged import PagedKVCacheManager
+
+
+def paged_generate(
```

Comment:
can we add a unit test for this function?

Reply:
yes, would require GPU and with known input/output. Does that work?

Reply:
why does it require GPU? in fms we have generate tests like this:

https://github.com/foundation-model-stack/foundation-model-stack/blob/main/tests/utils/test_generate.py#L27

Reply:
Currently, paged kernels only work on cuda

Reply:
added test which tests the output of a micro model in paged_generate vs a micro model with fms generate in latest commit